dnf/dnssec.py

```diff
@@ -265,12 +283,19 @@ def check_imported_keys_validity():
         keys = RpmImportedKeys._query_db_for_gpg_keys()
         logger.info(any_msg(_("Testing already imported keys for their validity.")))
         for key in keys:
-            result = DNSSECKeyVerification.verify(key)
+            try:
+                result = DNSSECKeyVerification.verify(key)
+            except DnssecError as e:
+                # Errors in this exception should not be fatal, print it and just continue
+                logger.exception("Exception raised in DNSSEC extension: email={}, exception={}"
+                                 .format(key.email, e))
```

Comment:
You need to call repr(e) here, otherwise `__str__ `method of exception is called instead of your `__repr__`  custom implementation. On top of that, `.format(e)` behaves differently on python3 (prints msg) and python2 (prints only the empty string)
.format(key.email, repr(e))